sdk/src/main/java/com/schibsted/security/strongbox/sdk/internal/config/credentials/ProfileCredentialProvider.java

```diff
@@ -0,0 +1,156 @@
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import com.amazonaws.services.securitytoken.model.AssumeRoleResult;
+import com.amazonaws.services.securitytoken.model.Credentials;
+import com.schibsted.security.strongbox.sdk.internal.RegionResolver;
+import com.schibsted.security.strongbox.sdk.internal.config.AWSConfigPropertyKey;
+import com.schibsted.security.strongbox.sdk.internal.config.ConfigProviderChain;
+import com.schibsted.security.strongbox.sdk.types.ClientConfiguration;
+import com.schibsted.security.strongbox.sdk.types.ProfileIdentifier;
+import com.schibsted.security.strongbox.sdk.types.arn.RoleARN;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.schibsted.security.strongbox.sdk.internal.ClientConfigurationHelper.transformAndVerifyOrThrow;
+
+/**
+ * @author stiankri
+ */
+public class ProfileCredentialProvider implements AWSCredentialsProvider {
+    private ClientConfiguration clientConfiguration;
+    private ProfileIdentifier profile;
+    private Supplier<MFAToken> mfaTokenSupplier;
+
+    public ProfileCredentialProvider(ClientConfiguration clientConfiguration, ProfileIdentifier profile, Supplier<MFAToken> mfaTokenSupplier) {
+        this.clientConfiguration = clientConfiguration;
+        this.profile = profile;
+        this.mfaTokenSupplier = mfaTokenSupplier;
+    }
+
+    @Override
+    public AWSCredentials getCredentials() {
+        return getCredentialsFromProfile(profile);
+    }
+
+    @Override
+    public void refresh() {
+
+    }
+
+    private AWSCredentials getCredentialsFromProfile(final ProfileIdentifier profile) {
+        ConfigProviderChain configProvider = new ConfigProviderChain();
+        if (!configProvider.hasConfig()) {
+            throw new IllegalStateException("When using '--profile', an AWS credentials or config file must be present");
+        }
+
+        Optional<RoleARN> roleToAssume = configProvider.getRoleArn(profile);
+        if (roleToAssume.isPresent()) {
+            return assumeRole(clientConfiguration, configProvider, profile, roleToAssume.get());
+        } else {
+            return getStaticCredentials(configProvider, profile);
+        }
+    }
+
+    /**
+     * Resolve AWS credentials based on MFA/Assume role
+     *
+     * We will assume that if mfa_serial is defined, then role_arn and source_profile also has to be specified.
+     *
+     * Please note that Strongbox differ from the AWS CLI in the following:
+     * AWS CLI: 'Note that configuration variables for using IAM roles can only be in the AWS CLI config file.'
+     * Strongbox: '--assume-role' can be specified explicitly
+     *
+     * https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#using-aws-iam-roles
+     */
+    private AWSCredentials assumeRole(ClientConfiguration clientConfiguration,
+                                              ConfigProviderChain configProvider,
```

Comment:
alignment seems a bit weird for the function parameters

Reply:
Yes, I will fix that.